src/lambda/handler-runner/invoke-runner/InvokeRunner.js

```diff
@@ -0,0 +1,41 @@
+import InvokeContainer from './InvokeContainer.js'
+
+export default class InvokeRunner {
+  #codeDir = null
+  #serverless = null
+  #env = null
+  #container = null
+
+  constructor(funOptions, serverless, env) {
+    const { codeDir, functionKey, handler, runtime } = funOptions
+
+    this.#codeDir = codeDir
+    this.#serverless = serverless
+    this.#env = env
+    this.#container = new InvokeContainer(
+      env,
+      functionKey,
+      handler,
+      runtime,
+      serverless,
+    )
+  }
+
+  cleanup() {
+    return undefined
+  }
+
+  // context will be generated in container
+  async run(event) {
+    if (!this.#container.isRunning) {
+      await this.#container.start('')
+    }
+
+    const result = await this.#container.request(
+      event.body ? JSON.parse(event.body) : event,
```

Comment:
Maybe it's need to check if `event.body` is a string before parsing it with `JSON.parse()`.

I noticed that if it's used with lambda integration (not lambda-proxy), the following error occurred.

```
offline: Failure: Unexpected token o in JSON at position 1
SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at InvokeRunner.run (/Users/zeroclock/development/src/github.com/zeroclock/lambda-rust-test/node_modules/serverless-offline/dist/lambda/handler-runner/invoke-runner/InvokeRunner.js:58:110)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    ...
```

This error seems that `event.body` is object.
It worked fine in both cases after modifying as below:

```js
  const result = await _classPrivateFieldLooseBase(this, _container)[_container].request(event.body ? (typeof event.body === 'string' ? JSON.parse(event.body) : event.body)  : event);
```

Reply:
Thanks for the suggestion! I will test it some more and update the PR accordingly.

Reply:
This makes sense but it looks like ESlint is choking on the double nested ternary expression here.  Could we refactor this and get this PR merged @EgorDm?

<img width="832" alt="Screen Shot 2021-04-23 at 00 42 28" src="https://user-images.githubusercontent.com/80178212/115819511-f8eefd80-a3cc-11eb-8322-c766f4cbdfe6.png">

https://travis-ci.com/github/dherault/serverless-offline/jobs/440478379#L254

Reply:
Thanks for reply. Missed that one. Should be fixed in the latest commit.